actors/builtin/init/init_actor_state.go

```diff
@@ -0,0 +1,70 @@
+package init
+
+import (
+	addr "github.com/filecoin-project/go-address"
+	cid "github.com/ipfs/go-cid"
+	errors "github.com/pkg/errors"
+	typegen "github.com/whyrusleeping/cbor-gen"
+
+	abi "github.com/filecoin-project/specs-actors/actors/abi"
+	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	autil "github.com/filecoin-project/specs-actors/actors/util"
+	adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+)
+
+type AddrKey = adt.AddrKey
+
+type InitActorState struct {
+	AddressMap  cid.Cid // HAMT[addr.Address]abi.ActorID
+	NextID      abi.ActorID
+	NetworkName string
+}
+
+func ConstructState(store adt.Store, networkName string) (*InitActorState, error) {
+	emptyMap, err := adt.MakeEmptyMap(store)
+	if err != nil {
+		return nil, err
+	}
+
+	return &InitActorState{
+		AddressMap:  emptyMap.Root(),
+		NextID:      abi.ActorID(builtin.FirstNonSingletonActorId),
+		NetworkName: networkName,
+	}, nil
+}
+
+// Resolves an address to an ID-address, if possible.
+// If the provided address is not in the table, it is returned as-is. This means that ID-addresses
+// (which should only appear as values, not keys) and singleton actor addresses pass through unchanged.
+func (s *InitActorState) ResolveAddress(store adt.Store, address addr.Address) (addr.Address, error) {
```

Comment:
Would it be worth checking if this (`address`) is an `ID` address before proceeding or is it not possible to reach that state?

Reply:
The pass-through behaviour is useful to a caller who just wants to end up with an ID-address one way or another.